File: Lab/Lab7_CPP/SRP.cpp
#include <iostream> // poprawnie; ale, dlaczego tylko 2 zadanie z pięciu?
#include <string>
#include <fstream>

using namespace std;

// Klasa użytkownika
class User {
private:
    string name;
    string email;
    string phoneNumber;

public:
    User(const string& name, const string& email, const string& phoneNumber) : name(name), email(email), phoneNumber(phoneNumber) {}

    string getName() const {
        return name;
    }

    string getEmail() const {
        return email;
    }

    string getPhoneNumber() const {
        return phoneNumber;
    }

    void setName(const string& newName) {
        name = newName;
    }

    void setEmail(const string& newEmail) {
        email = newEmail;
    }

    void setPhoneNumber(const string& newPhoneNumber) {
        phoneNumber = newPhoneNumber;
    }

};


// Klasa odpowiedzialna za zapis użytkownika
class UserSaver {
private:
    string File;

public:

    UserSaver(const string& File) : File(File) {}

    void saveUser(const User& user) {
        fstream file(File, ios::app);
        
        if (file.is_open()) {
            file << "Name: " << user.getName() << endl << "Email: " << user.getEmail() << endl << "Phone Number: " << user.getPhoneNumber() << "\n" << endl;
            file.close();
        } 
    }
};

int main() {
    
    User user("Jan Nowak", "jan.nowak@abc.pl", "123456789");

    User user2("Pawel Kowalski", "pawel.kowalski@abc.pl", "987654321");

    UserSaver saver("file1.txt");

    saver.saveUser(user);
    saver.saveUser(user2);
    return 0;
}

File: Lab/Lab7_Java/src/SRP.java
import java.io.FileWriter;  // poprawnie
import java.io.IOException;
import java.io.PrintWriter;

class User {
    private String name;
    private String email;
    private String phoneNumber;

    public User(String name, String email, String phoneNumber) {
        this.name = name;
        this.email = email;
        this.phoneNumber = phoneNumber;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setName(String newName) {
        name = newName;
    }

    public void setEmail(String newEmail) {
        email = newEmail;
    }

    public void setPhoneNumber(String newPhoneNumber) {
        phoneNumber = newPhoneNumber;
    }
}

class UserSaver {
    private String file;

    public UserSaver(String file) {
        this.file = file;
    }

    public void saveUser(User user) {
        try (FileWriter fw = new FileWriter(file, true);
             PrintWriter writer = new PrintWriter(fw)) {
            writer.println("Name: " + user.getName());
            writer.println("Email: " + user.getEmail());
            writer.println("Phone Number: " + user.getPhoneNumber());
            writer.println();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

public class SRP {
    public static void main(String[] args) {
        User user1 = new User("Jan Nowak", "jan.nowak@abc.pl", "123456789");
        User user2 = new User("Pawel Kowalski", "pawel.kowalski@abc.pl", "987654321");

        UserSaver saver = new UserSaver("file1.txt");

        saver.saveUser(user1);
        saver.saveUser(user2);
    }
}
